spinalcordtoolbox/scripts/sct_compute_compression.py

```diff
@@ -0,0 +1,631 @@
+        nargs=2,
+        metavar="[0 100]",
+        help='Age range of healthy subjects to use for the normalization. Requires the flag "-file-participants". '
+        'Example: "-age 60 80". By default, all ages are considered.'
+    )
+    optional.add_argument(
+        '-o',
+        metavar=Metavar.file,
+        help='Output CSV file name. If not provided, the suffix "_compression_metrics" is added to the file name provided by the flag "-i"'
+    )
+    optional.add_argument(
+        "-h",
+        "--help",
+        action="help",
+        help="Show this help message and exit")
+    optional.add_argument(
+        '-v',
+        metavar=Metavar.int,
+        type=int,
+        choices=[0, 1, 2],
+        default=1,
+        # Values [0, 1, 2] map to logging levels [WARNING, INFO, DEBUG], but are also used as "if verbose == #" in API
+        help="Verbosity. 0: Display only errors/warnings, 1: Errors/warnings + info messages, 2: Debug mode")
+
+    return parser
+
+
+# Functions for Step 1 (Load subject input files and get csv files)
+# ==========================================================================================
+def get_slice_thickness(img):
+    """
+    Get slice thickness from the input image.
+    :param img: Image: source image
+    :return float: slice thickness in mm.
+    """
+    return img.dim[6]
+
+
+def get_compressed_slice(img, verbose):
+    """
+    Get all the compression labels (voxels of value: '1') that are contained in the input image.
+    :param img: Image: source image
+    :return list: list of slices number
+    """
+    # Get all coordinates
+    coordinates = img.getNonZeroCoordinates(sorting='z')
+    logger.debug(f'Compression labels coordinates: {coordinates}')
+    # Check it coordinates is empty
+    if not coordinates:
+        raise ValueError('No compression labels found.')
+    # Return only slices number
+    return [int(coordinate.z) for coordinate in coordinates]
+
+
+def get_verterbral_level_from_slice(slices, df_metrics):
+    """
+    From slices, gets the coresponding vertebral level and creates a dict fo level and corresponding slice(s).
+    :param slices: list: list of slices number.
+    :param df_metrics: pandas.DataFrame: dataframe of metrics (output of sct_process_segmentation).
+    :return level_slice_dict: dict:
+    """
+    idx = df_metrics['Slice (I->S)'].isin(slices).tolist()
+    level_compression = df_metrics.loc[idx, ['VertLevel', 'Slice (I->S)']]
+    if level_compression.empty:
+        raise ValueError(f"Slice {slices} doesn't have a computed metric")
+    # Check if level_compression['VertLevel'] is nan
+    for _, row in level_compression.iterrows():
+        if np.isnan(row['VertLevel']):
+            raise ValueError(f"Slice {int(row['Slice (I->S)'])} doesn't have computed vertebral level. "
+                             f"Check vertebral labeling file.")
+    level_slice_dict = {}
+    for level in np.unique(level_compression['VertLevel']):
```

Comment:
This will ignore other slices in the same level that could refer to another compression. 

E.g.: If a compression is at C5/C6 and a second at C6/C7, bot compressions labels could be inside C6 at its extremities.

I need to adjust this. Also, we need to consider is multiple labels refer to the same compressions (the slices will be consecutive).

Reply:
Done in https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/4003/commits/074c19e51222294ee40d47fe32e325fd705b017f